plugin/plugin.go

```diff
@@ -0,0 +1,645 @@
+}
+
+type PluginConfigParams struct {
+	Level string `envconfig:"PLUGIN_LOG_LEVEL"`
+}
+
+type PluginInputParams struct {
+	Url                string `envconfig:"PLUGIN_URL"`
+	HttpMethod         string `envconfig:"PLUGIN_HTTP_METHOD"`
+	Headers            string `envconfig:"PLUGIN_HEADERS"`
+	ContentType        string `envconfig:"PLUGIN_CONTENT_TYPE"`
+	RequestBody        string `envconfig:"PLUGIN_REQUEST_BODY"`
+	AuthBasic          string `envconfig:"PLUGIN_AUTH_BASIC"`
+	AuthCert           string `envconfig:"PLUGIN_AUTH_CERT"`
+	ValidResponseCodes string `envconfig:"PLUGIN_VALID_RESPONSE_CODES"`
+	ValidResponseBody  string `envconfig:"PLUGIN_VALID_RESPONSE_BODY"`
+	Timeout            int    `envconfig:"PLUGIN_TIMEOUT"`
+	IgnoreSsl          bool   `envconfig:"PLUGIN_IGNORE_SSL"`
+	Proxy              string `envconfig:"PLUGIN_PROXY"`
+	OutputFile         string `envconfig:"PLUGIN_OUTPUT_FILE"`
+	AcceptType         string `envconfig:"PLUGIN_ACCEPT_TYPE"`
+	LogResponse        bool   `envconfig:"PLUGIN_LOG_RESPONSE"`
+	Quiet              bool   `envconfig:"PLUGIN_QUIET"`
+	UploadFile         string `envconfig:"PLUGIN_UPLOAD_FILE"`
+	MultiPartName      string `envconfig:"PLUGIN_MULTIPART_NAME"`
+	WrapAsMultipart    bool   `envconfig:"PLUGIN_WRAP_AS_MULTIPART"`
+	SslCertPath        string `envconfig:"PLUGIN_SSL_CERT_PATH"`
+}
+
+type PluginProcessingInfo struct {
+	AuthUser                 string
+	AuthPass                 string
+	HttpRequestCancelContext context.CancelFunc
+	BodyIoReader             io.Reader
+	HttpReq                  *http.Request
+	TimeOutDuration          time.Duration
+	httpClient               *http.Client
+	httpResponse             *http.Response
+	httpResponseBodyBytes    []byte
+	isConnectionOpen         bool
+	proxyUrl                 *url.URL
+	uploadFileAbsolutePath   string
+	IsSuppressLogs           bool
+}
+
+type PluginExecResultsCard struct {
+	ResponseStatus  int    `json:"RESPONSE_STATUS"`
+	ResponseContent string `json:"RESPONSE_CONTENT"`
+	ResponseHeaders string `json:"RESPONSE_HEADERS"`
+	ResponseFile    string `json:"RESPONSE_FILE"`
+}
+
+type Plugin struct {
+	Args
+	PluginProcessingInfo
+	PluginExecResultsCard
+}
+
+func Exec(ctx context.Context, args Args) error {
+
+	plugin := GetNewPlugin(args)
+
+	_ = plugin.Init()
+
+	err := plugin.Run()
+	if err != nil {
+		return err
+	}
+
+	err = plugin.DeInit()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func GetNewPlugin(args Args) *Plugin {
+	return &Plugin{
+		Args: args,
+	}
+}
+
+func (p *Plugin) Init() error {
+	return nil
+}
+
+func (p *Plugin) DeInit() error {
+
+	var err error
+
+	if p.isConnectionOpen {
+		p.isConnectionOpen = false
+		err = p.httpResponse.Body.Close()
+	}
+
+	if p.HttpRequestCancelContext != nil {
+		p.HttpRequestCancelContext()
+		p.HttpRequestCancelContext = nil
+	}
+
+	if p.BodyIoReader != nil {
+		if closer, ok := p.BodyIoReader.(io.Closer); ok {
+			err = closer.Close()
+		}
+	}
+
+	LogPrintln(p, "DeInit() called")
```

Comment:
```suggestion
	if err != nil {
		LogPrintln(p, "DeInit() called with error:", err)
	}
```

let's log this only when there's an error